### PR TITLE
Add new categories and uses to the Default Taxonomy

### LINKED ISSUE #141: Update fideslang data uses to support GVL
Attached updates: 

Let's figure out which migrations need to occur to support this.
[data_uses.csv](https://github.com/ethyca/fideslang/files/12134534/data_uses.csv)

### LINKED ISSUE #142: Update fideslang data categories to support GVL
Attached updated CSV. Lets discuss migration scripts.

[data_categories.csv](https://github.com/ethyca/fideslang/files/12147470/data_categories.csv)

### PR DESCRIPTION
**Pull request by ThomasLaPiana**
Closes #142 
Closes #141 

### Description Of Changes

[Link to Data Categories Sheet](https://docs.google.com/spreadsheets/d/1yFJ6TUPCRBd5Lqw3mI42hmRu93k9ZGkMG6RwL5dmAaI/edit?usp=sharing)
[Link to Data Uses Sheet](https://docs.google.com/spreadsheets/d/12wVsC4LSc-7MSiB-2v2mmSUIwEwWM8VBueIIn-VVZP0/edit?usp=sharing)

*Follow-Up Issue for Docs*: https://github.com/ethyca/fideslang/issues/150

__NOTE__: CI fixes happening here: https://github.com/ethyca/fideslang/pull/145

*All* of the changes we make here will need to get listed out for the corresponding migration PR over in `Fides`

### Code Changes

* [x] add new data categories
* [x] add new data uses
* [x] add additional assertions to the tests for additional data integrity checks

### Steps to Confirm

* [ ] _list any manual steps taken to confirm the changes_

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated (Authors note: New files generated for the docs site)
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
@NevilleS I'm thinking this needs a 2.0 bump? These are pretty large changes, and will definitely break things as we don't have proper versioning in yet

**Comment by ThomasLaPiana** [comment]
I'm investigating CI failures here due to our friend `PyYaml` and `Cython`: https://github.com/yaml/pyyaml/issues/736

**Comment by ThomasLaPiana** [comment]
@pattisdr will merging this hinder your in-flight work in any way?

**Comment by pattisdr** [comment]
Not sure, you seem to be working in different files though.  Either way we both have items that need to get in this sprint so I'd just go for it when you're ready -

File: src/fideslang/default_taxonomy/data_uses.py [hunk 1]
```diff
@@ -1,376 +1,359 @@
...
-        organization_fides_key="default_organization",
         name="Essential for Service Operations",
         description="Ensures the operation of the product, service, application or system.",
         parent_key="essential.service",
-        is_default=True,
     ),
     DataUse(
         fides_key="essential.service.operations.support",
-        organization_fides_key="default_organization",
         name="Essential for Serivce Operations Support",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py]: Existing typo Serivce but it also looks like the new name is supposed to be: Essential for Operations Support

File: src/fideslang/default_taxonomy/data_uses.py [hunk 2]
```diff
@@ -1,376 +1,359 @@
...
+    ),
+    DataUse(
+        fides_key="essential.service.operations.improve",
+        name="Essential for Support Improvement",
+        description="Essential to optimize and improve support for for the product, service, application or system.",
+        parent_key="essential.service.operations",
     ),
     DataUse(
         fides_key="essential.service.operations.support.optimization",
         name="Essential for Serivce Support Optimization",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py]: Existing typo Serivce

updated: looks like this whole data use can be removed?

File: data_files/data_uses.json [hunk 3]
```diff
@@ -337,12 +379,26 @@
             "is_default": true
         },
         {
-            "fides_key": "improve.system",
+            "fides_key": "functional.storage",
```
Comment (pattisdr) [inline data_files/data_uses.json]: This removed "improve.system" key is referenced in failing_nested_dataset.yml, system.md, and demo_system.yml

File: data_files/data_uses.yml [hunk 4]
```diff
@@ -287,24 +325,37 @@ data_use:
   legitimate_interest: false
   legitimate_interest_impact_assessment: null
   is_default: true
-- fides_key: improve
```
Comment (pattisdr) [inline data_files/data_uses.yml:290]: Former `improve` referenced in data_uses.md.  Actually that whole file will likely need a revamp at some point

File: src/fideslang/default_taxonomy/data_uses.py [hunk 5]
```diff
@@ -1,376 +1,359 @@
...
-        organization_fides_key="default_organization",
         name="Essential for Serivce Operations Support",
         description="Provides support for the product, service, application or system.",
         parent_key="essential.service.operations",
-        is_default=True,
+    ),
+    DataUse(
+        fides_key="essential.service.operations.improve",
+        name="Essential for Support Improvement",
+        description="Essential to optimize and improve support for for the product, service, application or system.",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py]: Duplicate words "for for"

File: src/fideslang/default_taxonomy/data_uses.py [hunk 6]
```diff
@@ -1,376 +1,359 @@
...
+        parent_key="essential.service.operations",
     ),
     DataUse(
         fides_key="essential.service.operations.support.optimization",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py]: Referencing [this doc](https://ethyca.atlassian.net/wiki/spaces/PM/pages/2740027499/Fideslang+updates#Taxonomy-updates-for-data-uses-%5BmediaInline%5D) - if current, it looks like this field should have been replaced by essential.service.operations.improve, but it's still here -

File: src/fideslang/default_taxonomy/data_uses.py [hunk 7]
```diff
@@ -1,376 +1,359 @@
...
-        organization_fides_key="default_organization",
         name="Essential SMS Service Notifications",
         description="Sends SMS notifications about the product, service, application or system.",
         parent_key="essential.service.notifications",
-        is_default=True,
     ),
     DataUse(
         fides_key="essential.service.operations",
-        organization_fides_key="default_organization",
         name="Essential for Service Operations",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py]: It looks like this should be the new name: Essential for Operations

File: src/fideslang/default_taxonomy/data_uses.py [hunk 8]
```diff
@@ -1,376 +1,359 @@
...
         description="Ensures the operation of the product, service, application or system.",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py]: It looks like this should be the new (similar) description: Essential to ensure the operation of the product, service, application or system.

File: src/fideslang/default_taxonomy/data_uses.py [hunk 9]
```diff
@@ -1,376 +1,359 @@
...
         fides_key="essential.service.operations.support.optimization",
         name="Essential for Serivce Support Optimization",
         description="Optimizes and improves support for the product, service, application or system.",
         parent_key="essential.service.operations.support",
-        organization_fides_key="default_organization",
-        is_default=True,
     ),
     DataUse(
         fides_key="essential.service.payment_processing",
         name="Essential for Service Payment Processing",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py]: It looks like new name should be: Essential for Payment Processing

File: src/fideslang/default_taxonomy/data_uses.py [hunk 10]
```diff
@@ -1,376 +1,359 @@
...
         description="Processes payments for the product, service, application or system.",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py]: It looks like new description should be similar: Essential to process payments for the product, service, application or system.

File: src/fideslang/default_taxonomy/data_uses.py [hunk 11]
```diff
@@ -1,376 +1,359 @@
...
-        organization_fides_key="default_organization",
         name="Train AI System",
         description="Trains an AI system or data model for machine learning.",
         parent_key=None,
-        is_default=True,
     ),
 ]
+
+for category in DEFAULT_DATA_USES:
+    category.is_default = True
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py:359]: Ah bravo!

File: src/fideslang/default_taxonomy/data_uses.py [hunk 12]
```diff
@@ -1,376 +1,359 @@
 from fideslang.models import DataUse
 
 DEFAULT_DATA_USES = [
-    # Analytics
+    #############
+    # Analytics #
+    #############
     DataUse(
         fides_key="analytics",
-        organization_fides_key="default_organization",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py:7]: Are we removing these because organizations have their own fides keys now?

File: src/fideslang/default_taxonomy/data_categories.py [hunk 13]
```diff
@@ -1,452 +1,604 @@
...
     ),
     DataCategory(
-        fides_key="user.payment.financial_account_number",
-        organization_fides_key="default_organization",
-        name="Account Payment Financial Account Number",
-        description="Financial account number for an account's payment card, bank account, or other financial system.",
-        parent_key="user.payment",
-        is_default=True,
+        fides_key="user.account.settings",
+        name="Account Information",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: copy/paste Should be Account Settings

File: src/fideslang/default_taxonomy/data_categories.py [hunk 14]
```diff
@@ -1,452 +1,604 @@
...
+        description="Scope of permissions and access to a system.",
         parent_key="user",
-        is_default=True,
     ),
     DataCategory(
-        fides_key="user.biometric_health",
-        organization_fides_key="default_organization",
-        name="Biometric Health Data",
-        description="Encoded characteristic collected about a user.",
+        fides_key="user.authorization.credentials",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py:41]: It looks like the fides_key for this one needs to be `user.authorization.credentials.unencrypted`, and we need to separately add `user.authorization.credentials`.

And if we decided to skip the proposed `user.authorization.credentials.unencrypted`, the description here needs to be updated instead.

File: src/fideslang/default_taxonomy/data_categories.py [hunk 15]
```diff
@@ -1,452 +1,604 @@
...
+        name="Account Credentials.",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: Doc has Account password here instead

File: src/fideslang/default_taxonomy/data_categories.py [hunk 16]
```diff
@@ -1,452 +1,604 @@
...
-        description="Demographic data about a user.",
+        fides_key="user.biometric.fingerprint",
+        name="Fingerprint",
+        description="Fingerprint encoded data about a subject.",
+        parent_key="user.biometric",
+    ),
+    DataCategory(
+        fides_key="user.biometric.retinal",
+        name="Retina Scan",
+        description="Retinal encoded data about a subject.",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: Is that really how you say that, "Retinal encoded?" Encoded via the retina? Just English question

File: src/fideslang/default_taxonomy/data_categories.py [hunk 17]
```diff
@@ -1,452 +1,604 @@
...
-        fides_key="user.social",
-        organization_fides_key="default_organization",
-        name="Social Data",
-        description="Social activity and interaction data.",
-        parent_key="user",
-        is_default=True,
+        fides_key="user.id.pseudonymous",
+        name="Pseudonymous User ID",
+        description="A pseudonymous, or probabilistic identifier generated from other subject or device data belonging to the subject.",
+        parent_key="user.id",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: It doesn't seem like there's a fides_key of `user.id` yet.  I see in a separate spreadsheet you're favoring unique_id but we need to address the parent key then

File: src/fideslang/default_taxonomy/data_categories.py [hunk 18]
```diff
@@ -1,452 +1,604 @@
...
+    DataCategory(
+        fides_key="user.location",
+        name="Location Data",
+        description="Records of the location of a user.",
+        parent_key="user",
+    ),
+    DataCategory(
+        fides_key="user.location.imprecise",
+        name="Imprecise Subject Location",
+        description="Imprecise location derived form sensors (more than 500M).",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: derived *from*

File: src/fideslang/default_taxonomy/data_categories.py [hunk 19]
```diff
@@ -1,452 +1,604 @@
...
+        parent_key="user.location",
+    ),
+    DataCategory(
+        fides_key="user.location.precise",
+        name="Precise Subject Location",
+        description="Precise location derived form sensors (less than 500M).",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: derived *from*

File: src/fideslang/default_taxonomy/data_categories.py [hunk 20]
```diff
@@ -1,452 +1,604 @@
...
     DataCategory(
         fides_key="user.non_specific_age",
-        organization_fides_key="default_organization",
         name="Non-Specific Age",
         description="Age range data.",
         parent_key="user",
-        is_default=True,
     ),
     DataCategory(
         fides_key="user.observed",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: This is supposed to be removed in favor of user.behavior

File: src/fideslang/default_taxonomy/data_categories.py [hunk 21]
```diff
@@ -1,452 +1,604 @@
...
-    DataCategory(
-        fides_key="user.media_consumption",
-        organization_fides_key="default_organization",
-        name="Media Consumption Data",
-        description="Media type consumption data of a user.",
-        parent_key="user",
-        is_default=True,
     ),
     DataCategory(
         fides_key="user.non_specific_age",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: Should be removed in favor of user.demographic.age_range

File: src/fideslang/default_taxonomy/data_categories.py [hunk 22]
```diff
@@ -1,452 +1,604 @@
...
-    DataCategory(
-        fides_key="user.credentials.password",
-        organization_fides_key="default_organization",
-        name="Password",
-        description="Password for system authentication.",
-        parent_key="user.credentials",
-        is_default=True,
     ),
     DataCategory(
         fides_key="user.date_of_birth",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: Should be removed in favor of user.demographic.date_of_birth

File: src/fideslang/default_taxonomy/data_categories.py [hunk 23]
```diff
@@ -1,452 +1,604 @@
...
+        description="Payment data related to user.",
         parent_key="user",
-        is_default=True,
     ),
     DataCategory(
-        fides_key="user.race",
-        organization_fides_key="default_organization",
-        name="Race",
-        description="Racial or ethnic origin data.",
+        fides_key="user.profiling",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: user.profiling should be removed in favor of user.demographic.profile

File: src/fideslang/default_taxonomy/data_categories.py [hunk 24]
```diff
@@ -1,452 +1,604 @@
...
+        name="Profiling Data",
+        description="Preference and interest data about a user.",
         parent_key="user",
-        is_default=True,
     ),
     DataCategory(
         fides_key="user.religious_belief",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: Should be removed in favor of user.demographic.religious_belief

File: src/fideslang/default_taxonomy/data_categories.py [hunk 25]
```diff
@@ -1,452 +1,604 @@
...
     DataCategory(
         fides_key="user.financial",
-        organization_fides_key="default_organization",
         name="Financial Data",
         description="Payment data and financial history.",
         parent_key="user",
-        is_default=True,
     ),
     DataCategory(
         fides_key="user.financial.account_number",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: Should be removed in favor of user.financial.bank_account

File: src/fideslang/default_taxonomy/data_categories.py [hunk 26]
```diff
@@ -1,452 +1,604 @@
...
+    # user.<misc> #
+    ###############
+    DataCategory(
+        fides_key="user.criminal_history",
+        name="Criminal History",
+        description="Criminal records or information about the data subject.",
+        parent_key="user",
+    ),
+    DataCategory(
+        fides_key="user.genetic",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py]: Should be removed in favor of user.health_and_medical.genetic

File: data_files/data_categories.csv [hunk 27]
```diff
@@ -1,58 +1,94 @@
-fides_key,is_default,name,organization_fides_key,parent_key,tags,description
```
Comment (pattisdr) [inline data_files/data_categories.csv:1]: What is the purpose of these csv/yml/and json files? Just curious how people use them

File: data_files/data_uses.json [hunk 28]
```diff
@@ -337,12 +379,26 @@
...
+            "is_default": true
+        },
+        {
+            "fides_key": "functional.service.improve",
+            "organization_fides_key": "default_organization",
+            "tags": null,
+            "name": "Improve Service",
             "description": "Improves the specific product, service, application or system.",
-            "parent_key": "improve",
+            "parent_key": "functional.service",
```
Comment (pattisdr) [inline data_files/data_uses.json:401]: Looks like we need to add "functional.service" - it's listed as a parent key, but the particular item doesn't exist.

**Review by pattisdr — CHANGES_REQUESTED** [review]
@ThomasLaPiana Large effort here!

Reviewed by comparing taxonomy files [to this confluence doc](https://ethyca.atlassian.net/wiki/spaces/PM/pages/2740027499/Fideslang+updates). I understand there have been other conversations since this doc and it is outdated in places so I'll leave the details for you to sort - 

Nice testing additions as well.  Another nice test to add might be one to make sure all the parents (and their parents, etc) exist.

File: src/fideslang/default_taxonomy/data_categories.py [hunk 29]
```diff
@@ -1,452 +1,604 @@
...
-        name="Payment Data",
-        description="Payment data related to user.",
+        fides_key="user.account",
+        name="Account Information.",
+        description="Account creation or registration information.",
         parent_key="user",
-        is_default=True,
     ),
     DataCategory(
-        fides_key="user.payment.financial_account_number",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py:21]: Where is it specified that we are removing user.payment.financial_account_number?

File: src/fideslang/default_taxonomy/data_categories.py [hunk 30]
```diff
@@ -1,452 +1,604 @@
...
+    ######################
+    DataCategory(
+        fides_key="user.authorization",
+        name="Authorization Information.",
+        description="Scope of permissions and access to a system.",
         parent_key="user",
-        is_default=True,
     ),
     DataCategory(
-        fides_key="user.biometric_health",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py:37]: Where is it specified that we are removing this one? user.biometric_health

File: src/fideslang/default_taxonomy/data_categories.py [hunk 31]
```diff
@@ -1,452 +1,604 @@
...
-        organization_fides_key="default_organization",
-        name="Biometric Health Data",
-        description="Encoded characteristic collected about a user.",
+        fides_key="user.authorization.credentials",
+        name="Account Credentials.",
+        description="Unencrypted authentication credentials to a system.",
+        parent_key="user.authorization",
+    ),
+    DataCategory(
+        fides_key="user.authorization.biometric",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_categories.py:47]: Where is this one coming from?

EDIT: found it in a google sheets conversation -

File: data_files/data_categories.csv (hunk 27 above)
Comment (ThomasLaPiana) [reply data_files/data_categories.csv:1]: They're used for our auto-generated docs as well as making it easier for non-engineers to more easily consume the default taxonomy

File: src/fideslang/default_taxonomy/data_uses.py (hunk 11 above)
Comment (ThomasLaPiana) [reply src/fideslang/default_taxonomy/data_uses.py:359]: makes the file a _little_ easier to parse visually without redundant fields! Way more changes coming here in the PR on top of this one...

File: src/fideslang/default_taxonomy/data_categories.py (hunk 16 above)
Comment (ThomasLaPiana) [reply src/fideslang/default_taxonomy/data_categories.py]: This doesn't sound right to me either...

"Retinal data" or "Encoded retinal data" makes more sense to me

File: src/fideslang/default_taxonomy/data_categories.py (hunk 18 above)
Comment (ThomasLaPiana) [reply src/fideslang/default_taxonomy/data_categories.py]: this is a really good place to use "code suggestions" :)

File: src/fideslang/default_taxonomy/data_categories.py (hunk 19 above)
Comment (ThomasLaPiana) [reply src/fideslang/default_taxonomy/data_categories.py]: code suggestion

File: src/fideslang/default_taxonomy/data_categories.py (hunk 30 above)
Comment (ThomasLaPiana) [reply src/fideslang/default_taxonomy/data_categories.py:37]: there is now a `user.biometric.health`

File: src/fideslang/default_taxonomy/data_categories.py (hunk 29 above)
Comment (ThomasLaPiana) [reply src/fideslang/default_taxonomy/data_categories.py:21]: `user.financial.account_number`

**Comment by ThomasLaPiana** [comment]
@pattisdr the review here is kind of confusing...

I was given a CSV file and told that it is the intended end-state of the taxonomy (one CSV for each type actually, categories and uses)

I converted those CSV files into google sheets and marked them up/had comments as per the linked sheets at the top of the PR

Anything that _isn't_ in that file, I took as an explicit flag to remove it from the current taxonomy. There was additional discussion in the comments around what to do with certain fields

Hugely appreciate the thorough review here! I know it's hard to follow, I want @rsilvery to chime in as well and confirm the CSV in this PR matches the requirements. This is very detail-oriented work and deserves a lot of eyes on to verify my changes!

File: data_files/data_uses.json (hunk 3 above)
Comment (ThomasLaPiana) [reply data_files/data_uses.json]: if the tests are passing, the name itself isn't really relevant. Tests shouldn't be tied to a specific version of the taxonomy (outside of the specific `default taxonomy` tests we have)

File: data_files/data_uses.yml (hunk 4 above)
Comment (ThomasLaPiana) [reply data_files/data_uses.yml:290]: Yeah I'll open a follow-up issue for a full audit for Fideslang 2.0

File: src/fideslang/default_taxonomy/data_uses.py (hunk 12 above)
Comment (ThomasLaPiana) [reply src/fideslang/default_taxonomy/data_uses.py:7]: no, it's because I didn't need to add it in the first place!
https://github.com/ethyca/fideslang/blob/main/src/fideslang/models.py#L69

File: CHANGELOG.md [hunk 32]
```diff
@@ -22,6 +22,10 @@ The types of changes are:
 
 - Add new fields to System and Privacy Declarations to support GVL [#146](https://github.com/ethyca/fideslang/pull/146)
 
+### Changed
+
+- Updated the Data Categories and Data Uses to support GVL [#144](https://github.com/ethyca/fideslang/pull/144)
+
```
Comment (pattisdr) [inline CHANGELOG.md]: Changelog will need to be updated now that we've released 1.4.4

File: src/fideslang/default_taxonomy/data_uses.py [hunk 33]
```diff
@@ -1,376 +1,353 @@
...
+        description="Stores or accesses information from the device as needed when using a product, service, application, or system",
+        parent_key="functional",
+    ),
+    DataUse(
+        fides_key="functional.service.improve",
+        name="Improve Service",
         description="Improves the specific product, service, application or system.",
-        parent_key="improve",
-        is_default=True,
+        parent_key="functional.service",
```
Comment (pattisdr) [inline src/fideslang/default_taxonomy/data_uses.py:192]: This parent key still does not exist in this taxonomy.

https://github.com/ethyca/fideslang/pull/144#discussion_r1289216214

I think a new test is useful at this point:
```python
    @pytest.mark.parametrize("data_type", ["data_category", "data_use"])
    def test_parent_key_exists(self, data_type) -> None:
        keys = [x.fides_key for x in getattr(DEFAULT_TAXONOMY, data_type)]
        for key in keys:
            expanded_keys: List[FidesKey] = []
            for i, split in enumerate(key.split(".")):
                expanded_keys.append(expanded_keys[-1] + "." + split if i else split)
                if expanded_keys[-1] not in keys:
                    raise Exception(
                        f"Missing parent key {expanded_keys[-1]} for {data_type}"
                    )
```

**Review by pattisdr — APPROVED** [review]
thanks for your changes here - approved with the addition of the missing parent fides key item

File: src/fideslang/default_taxonomy/data_uses.py (hunk 33 above)
Comment (ThomasLaPiana) [reply src/fideslang/default_taxonomy/data_uses.py:192]: This is super helpful! Next time feel free to contribute directly, I have short toes 🙂

**Comment by rsilvery** [comment]
Is there still something I need to review here? I don't get GH notifications in a timely way so please ping me on Slack if so.
